packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -3264,6 +3264,20 @@ describe("normalizeAndValidateConfig()", () => {
 			                    - The field \\"unsafe.metadata\\" should be an object but got null."
 		              `);
 			});
+
+			it("should not provide an unsafe warning when the environment variable is specified", () => {
+				process.env.WRANGLER_DISABLE_EXPERIMENTAL_WARNING = "1";
+
+				const { diagnostics } = normalizeAndValidateConfig(
+					{ unsafe: { bindings: [] } } as unknown as RawConfig,
+					undefined,
+					{ env: undefined }
+				);
+				delete process.env.WRANGLER_DISABLE_EXPERIMENTAL_WARNING;
```

Comment:
Ah - this probably should have been in a `try-finally` block in case the function call fails.